packages/language-service/ivy/completions.ts

```diff
@@ -211,6 +211,20 @@ export class CompletionBuilder<N extends TmplAstNode|AST> {
 
       const replacementSpan = makeReplacementSpanFromAst(this.node);
 
+      if (!(this.node.receiver instanceof ImplicitReceiver) &&
+          options?.includeCompletionsWithInsertText &&
+          options.includeAutomaticOptionalChainCompletions !== false) {
+        const symbol = this.templateTypeChecker.getSymbolOfNode(this.node.receiver, this.component);
+        if (symbol?.kind === SymbolKind.Expression) {
+          const type = symbol.tsType;
+          const nonNullableType = this.typeChecker.getNonNullableType(type);
+          if (type !== nonNullableType && replacementSpan !== undefined) {
```

Comment:
```suggestion
          if (type !== nonNullableType && replacementSpan !== undefined) {
            // Shift the start location back one so it includes the `.` of the property access. 
            //In combination with the options above, this will indicate to the TS LS to include optional chaining completions `?.` for nullable types.
```
Could you add a quick comment here just so it's easier for a passerby to understand why we're doing this?